test/parallel/test-runner-mock-timers.js

```diff
@@ -594,6 +594,185 @@ describe('Mock Timers Test Suite', () => {
+          });
+
+          t.mock.timers.tick(1000);
+          controller.abort();
+          t.mock.timers.tick(500);
+          t.mock.timers.tick(500);
+          t.mock.timers.tick(500);
+          await assert.rejects(() => p, {
+            name: 'AbortError',
+          });
+        });
+
+
+        it('should abort operation even if the .tick was not called', async (t) => {
+          t.mock.timers.enable({ apis: ['scheduler.wait'] });
+          const controller = new AbortController();
+          const p = nodeTimersPromises.scheduler.wait(2000, {
+            ref: true,
+            signal: controller.signal,
+          });
+
+          controller.abort();
+
+          await assert.rejects(() => p, {
+            name: 'AbortError',
+          });
+        });
+
+        it('should abort operation when .abort is called before calling setInterval', async (t) => {
+          t.mock.timers.enable({ apis: ['setTimeout'] });
+          const expectedResult = 'result';
+          const controller = new AbortController();
+          controller.abort();
+          const p = nodeTimersPromises.setTimeout(2000, expectedResult, {
+            ref: true,
+            signal: controller.signal,
+          });
+
+          await assert.rejects(() => p, {
+            name: 'AbortError',
+          });
+        });
+
+        it('should reject given an an invalid signal instance', async (t) => {
+          t.mock.timers.enable({ apis: ['scheduler.wait'] });
+          const p = nodeTimersPromises.scheduler.wait(2000, {
+            ref: true,
+            signal: {},
+          });
+
+          await assert.rejects(() => p, {
+            name: 'TypeError',
+            code: 'ERR_INVALID_ARG_TYPE',
+          });
+        });
+
+        // Test for https://github.com/nodejs/node/issues/50365
+        it('should not affect other timers when aborting', async (t) => {
+          const f1 = t.mock.fn();
+          const f2 = t.mock.fn();
+          t.mock.timers.enable({ apis: ['scheduler.wait'] });
+          const ac = new AbortController();
+
+          // id 1 & pos 1 in priority queue
+          nodeTimersPromises.scheduler.wait(100, { signal: ac.signal }).then(f1, f1);
+          // id 2 & pos 1 in priority queue (id 1 is moved to pos 2)
+          nodeTimersPromises.scheduler.wait(50).then(f2, f2);
+
+          ac.abort(); // BUG: will remove timer at pos 1 not timer with id 1!
+
+          t.mock.timers.runAll();
+          await nodeTimersPromises.setImmediate(); // let promises settle
+
+          // First scheduler.wait is aborted
+          assert.strictEqual(f1.mock.callCount(), 1);
+          assert.strictEqual(f1.mock.calls[0].arguments[0].code, 'ABORT_ERR');
+
+          // Second scheduler.wait should resolve, but never settles, because it was eronously removed by ac.abort()
+          assert.strictEqual(f2.mock.callCount(), 1);
+        });
+
+        // Test for https://github.com/nodejs/node/issues/50365
```

Comment:
Why is all of this commented out?

Reply:
it's a WIP, I'll remove it later when it's ready